core/chains/evm/forwarders/forwarder_manager.go

```diff
@@ -0,0 +1,308 @@
+	evmtypes "github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/authorized_forwarder"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/authorized_receiver"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/offchain_aggregator_wrapper"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/smartcontractkit/sqlx"
+)
+
+// Config encompasses config used by fwdmgr
+//go:generate mockery --recursive --name Config --output ./mocks/ --case=underscore --structname Config --filename config.go
+type Config interface {
+	LogSQL() bool
+}
+
+var authTopics = []common.Hash{
+	authorized_receiver.AuthorizedReceiverAuthorizedSendersChanged{}.Topic(),
+	offchain_aggregator_wrapper.OffchainAggregatorConfigSet{}.Topic(),
+}
+
+var forwardABI = evmtypes.MustGetABI(authorized_forwarder.AuthorizedForwarderABI).Methods["forward"]
+
+var authLogCheckInterval = time.Duration(5 * time.Second)
+
+type FwdMgr struct {
+	ORM       ORM
+	config    Config
+	evmClient evmclient.Client
+	logger    logger.SugaredLogger
+	logpoller *evmlogpoller.LogPoller
+
+	// TODO(samhassan): sendersCache should be an LRU capped cache
+	sendersCache map[common.Address][]common.Address
+	latestBlock  int64
+
+	authRcvr    authorized_receiver.AuthorizedReceiverInterface
+	offchainAgg offchain_aggregator_wrapper.OffchainAggregatorInterface
+
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	cacheMu sync.RWMutex
+	chStop  chan struct{}
+	wg      sync.WaitGroup
+	utils.StartStopOnce
+}
+
+func NewFwdMgr(db *sqlx.DB, cfg Config, client evmclient.Client, logpoller *evmlogpoller.LogPoller, l logger.Logger) *FwdMgr {
+	lggr := logger.Sugared(l.Named("FwdMgr"))
+	fwdMgr := FwdMgr{
+		logger:       lggr,
+		evmClient:    client,
+		ORM:          NewORM(db, lggr, cfg),
+		logpoller:    logpoller,
+		config:       cfg,
+		sendersCache: make(map[common.Address][]common.Address),
+		chStop:       make(chan struct{}),
+		cacheMu:      sync.RWMutex{},
+		wg:           sync.WaitGroup{},
+		latestBlock:  0,
+	}
+	return &fwdMgr
+}
+
+// Start starts Forwarder Manager.
+func (f *FwdMgr) Start() error {
+	return f.StartOnce("EVMForwarderManager", func() error {
+		f.logger.Debug("Initializing EVM forwarder manager")
+
+		fwdrs, cnt, err := f.ORM.FindForwardersByChain(utils.Big(*f.evmClient.ChainID()))
+		if err != nil {
+			return errors.Wrapf(err, "Failed to retrieve forwarders for chain %d", f.evmClient.ChainID())
+		}
+		f.ctx, f.cancel = context.WithCancel(context.Background())
+		if cnt != 0 {
+			f.initForwardersCache(f.ctx, fwdrs)
+			f.subscribeForwardersLogs(fwdrs)
+		}
+
+		f.authRcvr, err = authorized_receiver.NewAuthorizedReceiver(common.Address{}, f.evmClient)
+		if err != nil {
+			f.logger.Panic("Failed to init AuthorizedReceiver")
```

Comment:
Probably should include the specific errors?
```suggestion
			f.logger.Panicw("Failed to init AuthorizedReceiver", "err", err)
```

Reply:
Returned wrapped error instead of panicing as connor suggested, seemed more graceful